src/components/AsyncDropdown/index.js

```diff
@@ -20,56 +21,88 @@ export default ({
   autocompleteRequest,
   placeholder,
   requestUrl
-}: Props) => (
-  <AsyncSelect
-    placeholder={answer || placeholder}
-    className="material-survey-selection"
-    styles={{
-      menu: provided => ({
-        ...provided,
-        fontFamily: "Roboto, sans-serif"
-      }),
-      container: provided => ({
-        ...provided,
-        fontFamily: "Roboto, sans-serif"
-      }),
-      menuPortal: provided => ({ ...provided, zIndex: 10000 })
-    }}
-    loadOptions={async inputValue => {
-      if (autocompleteRequest) {
-        return (await autocompleteRequest(requestUrl, inputValue)).map(
-          ({ value, label, subLabel }: any) => ({
-            value,
-            label: subLabel ? (
-              <div
-                aria-label={label}
-              >
-                {label}
+}: Props) => {
+  const MUItheme = useTheme()
+  return (
+    <AsyncSelect
+      placeholder={answer || placeholder}
+      className="material-survey-selection"
+      styles={{
+        menu: provided => ({
+          ...provided,
+          fontFamily: "Roboto, sans-serif"
+        }),
+        container: provided => ({
+          ...provided,
+          fontFamily: "Roboto, sans-serif"
+        }),
+        menuPortal: provided => ({ ...provided, zIndex: 10000 }),
+        singleValue: provided => ({
+          ...provided,
+          color: MUItheme.palette.text.primary
+        }),
+        option: (provided, { data, isDisabled, isFocused, isSelected }) => ({
+          ...provided,
+          color: (isFocused || isSelected) ? MUItheme.palette.primary.contrastText : MUItheme.palette.text.primary
+        })
+      }}
+      theme={theme => ({
+        ...theme,
+        colors: {
+          ...theme.colors,
+          primary: MUItheme.palette.primary.main,
+          primary75: MUItheme.palette.primary.light,
+          primary50: MUItheme.palette.primary.light,
+          primary25: MUItheme.palette.primary.light,
+          neutral0: MUItheme.palette.background.paper,
+          neutral5: MUItheme.palette.grey['50'],
+          neutral10: MUItheme.palette.grey['100'],
+          neutral20: MUItheme.palette.grey['200'],
+          neutral30: MUItheme.palette.grey['300'],
+          neutral40: MUItheme.palette.grey['400'],
+          neutral50: MUItheme.palette.grey['500'],
+          neutral60: MUItheme.palette.grey['600'],
+          neutral70: MUItheme.palette.grey['700'],
+          neutral80: MUItheme.palette.grey['800'],
+          neutral90: MUItheme.palette.grey['900']
+        }
+      })}
```

Comment:
this should be part of a global context or in a single file, in react when you define objects in component props it causes a lot of rerenders- it COULD be a problem from a performance perspective and it's a problem from a DRY perspective.

Reply:
I understand your concern, however, the theme really would change on some rare occasions, for example, switching between light mode and dark mode.

Is that okay, if I somehow "memorise" the `colors` object, therefore reduces the re-rendering of the object?

Reply:
```
const themeColors = useMemo(() => ({
  primary: MUItheme.palette.primary.main,
  primary75: MUItheme.palette.primary.light,
  primary50: MUItheme.palette.primary.light,
  primary25: MUItheme.palette.primary.light,
  neutral0: MUItheme.palette.background.paper,
  neutral5: MUItheme.palette.grey['50'],
  neutral10: MUItheme.palette.grey['100'],
  neutral20: MUItheme.palette.grey['200'],
  neutral30: MUItheme.palette.grey['300'],
  neutral40: MUItheme.palette.grey['400'],
  neutral50: MUItheme.palette.grey['500'],
  neutral60: MUItheme.palette.grey['600'],
  neutral70: MUItheme.palette.grey['700'],
  neutral80: MUItheme.palette.grey['800'],
  neutral90: MUItheme.palette.grey['900']
}), [MUItheme])
```

```
<AsyncSelect
  ...
  theme={theme => ({
    ...theme,
    colors: {
      ...theme.colors,
      ...themeColors
    }
  })}
  ...
```